Minor display fixes

michaelaoash opened the pull request:
Changed 3 display issues
(1) Changed "pi" to "π"
(2) Fixed bug in lRegisterDisplay of EE.  Was "# y EE x #", that is with the letters "y" and "x" printed.  Changed to "#EE#".
(3) Added scientific display for small, nonzero numbers.  This is mainly an issue for log10, which is often used on very small decimals, for example in the context of pH (acidity) in chemistry.  It is common to compute -log10(0.0000001) = 7 (for a neutral, neither acidic nor basic, solution).  So it would be nice to include that in scientific notation instead of displaying 0. **I'm not sure if the code will work.** 

BTW 1.91 (1) is superb. Don't change a thing. Thanks, as always.

michaelaoash committed:
Added gitignore and extra directory

michaelaoash committed:
Added gitignore and extra directory

michaelaoash committed:
Added gitignore and extra directory

michaelaoash committed:
Changed 3 display items: pi to π, EE output display bug, scientific notation for very small numbers

michaelaoash committed:
Changed 3 display items: pi to π, EE output display bug, scientific notation for very small numbers

premier-education commented:
Dear Michael,

Great, thanks - will have a look at the below. Just finished work (we do Sunday - Thursday in the UAE), time for the weekend and some coding!

I also sent through a link to Andrew, hopefully he sees it - the server came back up yesterday.

Hope you have been well.

Best,
Ashok

________________________________
From: Michael Ash <notifications@github.com>
Sent: Thursday, 23 April 2020 6:26 PM
To: premier-education/RPN-30 <RPN-30@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
Subject: [premier-education/RPN-30] Minor display fixes (#13)


Changed 3 display issues
(1) Changed "pi" to "π"
(2) Fixed bug in lRegisterDisplay of EE. Was "# y EE x #", that is with the letters "y" and "x" printed. Changed to "#EE#".
(3) Added scientific display for small, nonzero numbers. This is mainly an issue for log10, which is often used on very small decimals, for example in the context of pH (acidity) in chemistry. It is common to compute -log10(0.0000001) = 7 (for a neutral, neither acidic nor basic, solution). So it would be nice to include that in scientific notation instead of displaying 0. I'm not sure if the code will work.

BTW 1.91 (1) is superb. Don't change a thing. Thanks, as always.

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/premier-education/RPN-30/pull/13<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fpremier-education%2FRPN-30%2Fpull%2F13&data=02%7C01%7C%7C405ad06ad623444b66bd08d7e7600e8d%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637232272051735320&sdata=0IYokn7%2FFzHz9qQC6nJFVQEb2zaMWQ6MneNpqlPaP9g%3D&reserved=0>

Commit Summary

  *   Added gitignore and extra directory
  *   Added gitignore and extra directory
  *   Added gitignore and extra directory
  *   Changed 3 display items: pi to π, EE output display bug, scientific notation for very small numbers

File Changes

  *   A .gitignore<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fpremier-education%2FRPN-30%2Fpull%2F13%2Ffiles%23diff-a084b794bc0759e7a6b77810e01874f2&data=02%7C01%7C%7C405ad06ad623444b66bd08d7e7600e8d%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637232272051745315&sdata=Y7Sf8AQu8NPjKXm%2FV4L0hIzRm%2BtoP9huXkeYumTLst8%3D&reserved=0> (2)
  *   M RPN-30/Calculator.swift<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fpremier-education%2FRPN-30%2Fpull%2F13%2Ffiles%23diff-64eb0f1efb575385a10b5d37c68e54e5&data=02%7C01%7C%7C405ad06ad623444b66bd08d7e7600e8d%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637232272051745315&sdata=oDeCCwXsJedlRJHR0%2BmJbyAZNCI4OV7sqlIWNkNwlqQ%3D&reserved=0> (3)
  *   M RPN-30/LogicEngine.swift<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fpremier-education%2FRPN-30%2Fpull%2F13%2Ffiles%23diff-026f0dec18ae7fc63099e6ede91fbe5e&data=02%7C01%7C%7C405ad06ad623444b66bd08d7e7600e8d%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637232272051755312&sdata=pTvBxAo9DawboBMtUfkUXyBo7EsJghgroLmtl1%2FeutE%3D&reserved=0> (4)
  *   M RPN-30/OutputDisplay.swift<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fpremier-education%2FRPN-30%2Fpull%2F13%2Ffiles%23diff-6c046d00fcb7d9296d54f38df9e97e7e&data=02%7C01%7C%7C405ad06ad623444b66bd08d7e7600e8d%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637232272051755312&sdata=gBdfxmdGCkkQ3SvTxasiyvga%2FCzJ2XyhlLz1ylQWMZ0%3D&reserved=0> (16)

Patch Links:

  *   https://github.com/premier-education/RPN-30/pull/13.patch<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fpremier-education%2FRPN-30%2Fpull%2F13.patch&data=02%7C01%7C%7C405ad06ad623444b66bd08d7e7600e8d%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637232272051755312&sdata=MHgXTNv%2BiMDre6O2ILj45Pd5d2B%2BTO2pc%2FRPdZWbFpI%3D&reserved=0>
  *   https://github.com/premier-education/RPN-30/pull/13.diff<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fpremier-education%2FRPN-30%2Fpull%2F13.diff&data=02%7C01%7C%7C405ad06ad623444b66bd08d7e7600e8d%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637232272051765304&sdata=GAx%2Bf2hBpO%2BW2oK7qPw5hVUO1s4O4U9JzAOMT3LvYA8%3D&reserved=0>

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fpremier-education%2FRPN-30%2Fpull%2F13&data=02%7C01%7C%7C405ad06ad623444b66bd08d7e7600e8d%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637232272051765304&sdata=3MysLnqXbqbYs5LUuvEmIcqb7PuHiltg7vGjqCs9%2Fyw%3D&reserved=0>, or unsubscribe<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FALEWW2Z32NORFAIPBUXRMELRN73UHANCNFSM4MO3JSQA&data=02%7C01%7C%7C405ad06ad623444b66bd08d7e7600e8d%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637232272051775297&sdata=gfpOk913JEbPtxgEgI3A9Ro1G%2FBRS7hF3L%2FrNIOWCxM%3D&reserved=0>.

michaelaoash committed:
Merge branch 'minor-display-fixes' of https://github.com/michaelaoash/RPN-30 into minor-display-fixes

michaelaoash commented:
I further adjusted the displays so that very small decimals are shown in scientific notation.  The xRegister still needs adjustment.

premier-education commented:
Dear Michael,

Hope you are well.

I've made some further changes, including changing pi and then EE to "e", i.e. it will display like 55.234e123.

In addition, I've added in a "settings app", which can be accessed via going to settings on your iPhone and scrolling down to RPN-30. Herein, you can select which calculator configuration you want (you have to toggle the "More Scientific Functions" to get the layout in 1.91 that you were seeing - only a one off exercise). You can also choose to enable scientific mode (i.e. significant digits) and select how many digits you want to display. And also select how many decimals to round to if in decimal mode. For any changes to the settings app, typically you have to restart the calculator by swiping up and then re-opening. I think you will find scientific mode beneficial for your log10 query, however I will need to think of a easy way to switch between decimal and scientific within the app.

I've also fixed a display bug that I saw for the first time, but I noticed existed in the original app, which was a bit worrying. Related to a flaw / bug in Apple's code for number formatting.

I think this version might have a few bugs that need to be sorted out, hopefully will get some time on Tuesday to do that. Let me know if you find any and apologies for any inconvenience.

Best,
Ashok

michaelaoash commented:
Dear Ashok,

That sounds great and I look forward to the next round of testing.   I very
much like the large numbers getting smaller and smaller in font size
(better than scientific notation for the main display). I hope that will
remain -- I thought it was a very nice feature.  Anyway, I'm looking
forward to the next Test Flight.  Cool.

Also, I was able to install XCode on a Mac that my wife uses; so I can now
compile the code and run the simulator, which makes development easier.

Thanks again.

Yours,
Michael



Yours,
Michael




On Sun, Apr 26, 2020 at 4:40 PM Ashok Khanna <notifications@github.com>
wrote:

> Dear Michael,
>
> Hope you are well.
>
> I've made some further changes, including changing pi and then EE to "e",
> i.e. it will display like 55.234e123.
>
> In addition, I've added in a "settings app", which can be accessed via
> going to settings on your iPhone and scrolling down to RPN-30. Herein, you
> can select which calculator configuration you want (you have to toggle the
> "More Scientific Functions" to get the layout in 1.91 that you were seeing
> - only a one off exercise). You can also choose to enable scientific mode
> (i.e. significant digits) and select how many digits you want to display.
> And also select how many decimals to round to if in decimal mode. For any
> changes to the settings app, typically you have to restart the calculator
> by swiping up and then re-opening. I think you will find scientific mode
> beneficial for your log10 query, however I will need to think of a easy way
> to switch between decimal and scientific within the app.
>
> I've also fixed a display bug that I saw for the first time, but I noticed
> existed in the original app, which was a bit worrying. Related to a flaw /
> bug in Apple's code for number formatting.
>
> I think this version might have a few bugs that need to be sorted out,
> hopefully will get some time on Tuesday to do that. Let me know if you find
> any and apologies for any inconvenience.
>
> Best,
> Ashok
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/premier-education/RPN-30/pull/13#issuecomment-619621738>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFZJT24E65EB5AELRMPBOTROSL4XANCNFSM4MO3JSQA>
> .
>


-- 
Michael Ash, Professor of Economics and Public Policy
Member of MSP/MTA/NEA, the union of faculty and librarians
University of Massachusetts Amherst
Email mash@econs.umass.edu
Tel +1-413-545-6329

premier-education commented:
That’s great! I am a big fan of XCode (with the exception of Canvas UI tool) (I have spent the last week doing javascript in a web browser without any color coding, indents, ability to view the code while testing, it’s been a complete nightmare but a necessary one for Browser apps - the downside of Xcode / Swift is that it is Mac only).

The numbers getting smaller will remain as long as you don’t change to scientific mode. It is definitely a great feature, which I believe is Automatic from Apple’s side without any additional coding.

I think Scientific notation could also work, but the current set up is not optimized for it - the main display is centred and looks very nice as numbers flow to it; whereas scientific mode is a lot more rigid, so the display should reflect that as well (e.g. having six boxes for the six significant digits).

________________________________
From: Michael Ash <notifications@github.com>
Sent: Monday, April 27, 2020 1:11:33 AM
To: premier-education/RPN-30 <RPN-30@noreply.github.com>
Cc: Ashok Khanna <ashok.khanna@hotmail.com>; Comment <comment@noreply.github.com>
Subject: Re: [premier-education/RPN-30] Minor display fixes (#13)


Dear Ashok,

That sounds great and I look forward to the next round of testing. I very
much like the large numbers getting smaller and smaller in font size
(better than scientific notation for the main display). I hope that will
remain -- I thought it was a very nice feature. Anyway, I'm looking
forward to the next Test Flight. Cool.

Also, I was able to install XCode on a Mac that my wife uses; so I can now
compile the code and run the simulator, which makes development easier.

Thanks again.

Yours,
Michael



Yours,
Michael




On Sun, Apr 26, 2020 at 4:40 PM Ashok Khanna <notifications@github.com>
wrote:

> Dear Michael,
>
> Hope you are well.
>
> I've made some further changes, including changing pi and then EE to "e",
> i.e. it will display like 55.234e123.
>
> In addition, I've added in a "settings app", which can be accessed via
> going to settings on your iPhone and scrolling down to RPN-30. Herein, you
> can select which calculator configuration you want (you have to toggle the
> "More Scientific Functions" to get the layout in 1.91 that you were seeing
> - only a one off exercise). You can also choose to enable scientific mode
> (i.e. significant digits) and select how many digits you want to display.
> And also select how many decimals to round to if in decimal mode. For any
> changes to the settings app, typically you have to restart the calculator
> by swiping up and then re-opening. I think you will find scientific mode
> beneficial for your log10 query, however I will need to think of a easy way
> to switch between decimal and scientific within the app.
>
> I've also fixed a display bug that I saw for the first time, but I noticed
> existed in the original app, which was a bit worrying. Related to a flaw /
> bug in Apple's code for number formatting.
>
> I think this version might have a few bugs that need to be sorted out,
> hopefully will get some time on Tuesday to do that. Let me know if you find
> any and apologies for any inconvenience.
>
> Best,
> Ashok
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/premier-education/RPN-30/pull/13#issuecomment-619621738>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFZJT24E65EB5AELRMPBOTROSL4XANCNFSM4MO3JSQA>
> .
>


--
Michael Ash, Professor of Economics and Public Policy
Member of MSP/MTA/NEA, the union of faculty and librarians
University of Massachusetts Amherst
Email mash@econs.umass.edu
Tel +1-413-545-6329

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fpremier-education%2FRPN-30%2Fpull%2F13%23issuecomment-619625491&data=02%7C01%7C%7C295fdbfec63d42bf4c4108d7ea266672%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637235322950933834&sdata=OpY2jIsxZlow2xMdEu33v5xciewjuV1m31k7AHmKVx8%3D&reserved=0>, or unsubscribe<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FALEWW24DRLKE55XO4QQAUVDROSPQLANCNFSM4MO3JSQA&data=02%7C01%7C%7C295fdbfec63d42bf4c4108d7ea266672%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C637235322950938824&sdata=2Lpkb8P1ua6ug9XGMh8yXLeB0SX2NIcxLrEAp8KTmQM%3D&reserved=0>.
